src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/BlobListenerFactory.cs

```diff
@@ -198,50 +197,103 @@ internal class BlobListenerFactory : IListenerFactory
                 // Create a singleton shared blob listener, since we only
                 // want a single instance of the blob poll/scan logic to be running
                 // across host instances
-                var singletonBlobListener = _singletonManager.CreateHostSingletonListener(
-                    new BlobListener(sharedBlobListener), SingletonBlobListenerScopeId);
+                var singletonBlobListener = _singletonManager.CreateHostSingletonListener(sharedBlobListener, SingletonBlobListenerScopeId);
                 _sharedContextProvider.SetValue(SingletonBlobListenerScopeId, true);
-
-                return new CompositeListener(singletonBlobListener, queueListener);
+                // since now that queuelistener is null, we no longer need CompositeListener
+                return singletonBlobListener;
             }
             else
             {
                 // We've already created the singleton blob listener
                 // so just return our queue listener. Note that while we want the
                 // blob scan to be singleton, the shared queue listener needs to run
                 // on ALL instances so load can be scaled out
-                return queueListener;
+                return new NullListener();
             }
         }
 
         private async Task RegisterWithSharedBlobListenerAsync(
             string hostId,
             SharedBlobListener sharedBlobListener,
             IStorageBlobClient blobClient,
-            IStorageQueue hostBlobTriggerQueue,
-            IMessageEnqueuedWatcher messageEnqueuedWatcher,
+            DispatchProcessor dispatchProcessor,
             CancellationToken cancellationToken)
         {
             BlobTriggerExecutor triggerExecutor = new BlobTriggerExecutor(hostId, _functionId, _input,
-                BlobETagReader.Instance, new BlobReceiptManager(blobClient),
-                new BlobTriggerQueueWriter(hostBlobTriggerQueue, messageEnqueuedWatcher));
```

Comment:
The previous queue writer took the "messageEnqueuedWatcher" so that it could notify the queue listener quickly when messages were added. Are you retaining that capability? It allows the listener to break out quickly from its polling loop.

Reply:
yes, the logic is applied when we initialize the sharedQ
```
            SharedQueueWatcher sharedQueueWatcher = _sharedContextProvider.GetOrCreateInstance<SharedQueueWatcher>(
                                                        new SharedQueueWatcherFactory(_messageEnqueuedWatcherSetter));
            _sharedQueueWriter = new SharedQueueWriter(sharedQueue, sharedQueueWatcher);
```